libs/deal-ticket/src/components/deal-ticket/deal-ticket-stop-order.tsx

```diff
@@ -0,0 +1,600 @@
+  timeInForce: Schema.OrderTimeInForce.TIME_IN_FORCE_FOK,
+  triggerType: 'price',
+  triggerDirection:
+    Schema.StopOrderTriggerDirection.TRIGGER_DIRECTION_RISES_ABOVE,
+  expiryStrategy: Schema.StopOrderExpiryStrategy.EXPIRY_STRATEGY_SUBMIT,
+  size: '0',
+};
+
+const stopSubmit: FormEventHandler = (e) => e.preventDefault();
+
+export const StopOrder = ({ market, marketPrice, submit }: StopOrderProps) => {
+  const { pubKey, isReadOnly } = useVegaWallet();
+  const setDealTicketType = useDealTicketTypeStore((state) => state.set);
+  const [, updateOrder] = useOrder(market.id);
+  const updateStoredFormValues = useStopOrderFormValues(
+    (state) => state.update
+  );
+  const storedFormValues = useStopOrderFormValues(
+    (state) => state.formValues[market.id]
+  );
+  const { handleSubmit, setValue, watch, control, formState } =
+    useForm<StopOrderFormValues>({
+      defaultValues: { ...defaultValues, ...storedFormValues },
+    });
+  const { errors } = formState;
+  const lastSubmitTime = useRef(0);
+  const onSubmit = useCallback(
+    (data: StopOrderFormValues) => {
+      const now = new Date().getTime();
+      if (lastSubmitTime.current && now - lastSubmitTime.current < 1000) {
+        return;
+      }
+      submit(
+        mapFormValuesToStopOrdersSubmission(
+          data,
+          market.id,
+          market.decimalPlaces,
+          market.positionDecimalPlaces
+        )
+      );
+      lastSubmitTime.current = now;
+    },
+    [market.id, market.decimalPlaces, market.positionDecimalPlaces, submit]
+  );
+  const side = watch('side');
+  const expire = watch('expire');
+  const triggerType = watch('triggerType');
+  const triggerPrice = watch('triggerPrice');
+  const timeInForce = watch('timeInForce');
+  const type = watch('type');
+  const rawPrice = watch('price');
+  const rawSize = watch('size');
+
+  if (storedFormValues?.size && rawSize !== storedFormValues?.size) {
+    setValue('size', storedFormValues.size);
+  }
+  if (storedFormValues?.price && rawPrice !== storedFormValues?.price) {
+    setValue('price', storedFormValues.price);
+  }
+
+  const isPriceTrigger = triggerType === 'price';
+  const size = removeDecimal(rawSize, market.positionDecimalPlaces);
+  const price = useMemo(() => {
+    return (
+      marketPrice &&
+      getDerivedPrice(
+        {
+          type,
+          price: rawPrice && removeDecimal(rawPrice, market.decimalPlaces),
+        },
+        type === Schema.OrderType.TYPE_MARKET && isPriceTrigger && triggerPrice
+          ? removeDecimal(triggerPrice, market.decimalPlaces)
+          : marketPrice
+      )
+    );
+  }, [
+    market.decimalPlaces,
+    marketPrice,
+    rawPrice,
+    isPriceTrigger,
+    triggerPrice,
+    type,
+  ]);
+
+  const notionalSize = useNotionalSize(
+    price,
+    size,
+    market.decimalPlaces,
+    market.positionDecimalPlaces
+  );
+
+  useEffect(() => {
+    const subscription = watch((value, { name, type }) => {
```

Comment:
In the other deal ticket we have the reverse approach. Where the zustand store is the 'source of truth' for deal ticket data. I'd prefer it this way around, but perhaps we refactor in the follow up tickets

Reply:
I wanted to refactor it to make to not write through zustand store

Reply:
Lets not block over this. We'll address in the follow up tickets